core/src/test/java/com/linecorp/armeria/common/FlagsTest.java

```diff
@@ -48,12 +61,69 @@ public void epollAvailableOnLinux() {
      * on Linux, Windows and OS X.
      */
     @Test
-    public void openSslAvailable() {
+    void openSslAvailable() {
         assumeTrue(osName.startsWith("linux") || osName.startsWith("windows") ||
                    osName.startsWith("macosx") || osName.startsWith("osx"));
         assumeFalse("false".equals(System.getProperty("com.linecorp.armeria.useOpenSsl")));
 
         assertThat(Flags.useOpenSsl()).isTrue();
         assertThat(OpenSsl.isAvailable()).isTrue();
     }
+
+    @Test
+    void dumpOpenSslInfoDoNotThrowStackOverFlowError() throws Throwable {
+        assumeTrue(OpenSsl.isAvailable());
+        System.setProperty("com.linecorp.armeria.dumpOpenSslInfo", "true");
+
+        // There's a chance that Flags.useOpenSsl() is already called by other test cases, which means that
+        // we cannot set dumpOpenSslInfo. So we use our own class loader to load the Flags class.
```

Comment:
nit: The current workaround is not bad. But how about making `useOpenSsl` and `dumpOpenSslInfo` package-private? 
I think We can test `Flags.useOpenSsl()` after setting `null` to `useOpenSsl` and `dumpOpenSslInfo`.

Reply:
> I think We can test Flags.useOpenSsl() after setting useOpenSsl and dumpOpenSslInfo to null.

Yes, we can do that way as well. But I'm not sure if it's a good way. 🤔 

> The current workaround is not bad.

So you think setting `null` to the field and calling the method again is better?

Reply:
I guess the two options have both pros and cons. I'm fine with the class loader version. 😀